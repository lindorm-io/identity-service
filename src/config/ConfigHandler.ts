import { ConfigurationBase, IConfigurationDataBase, IConfigurationOptions } from "@lindorm-io/koa-config";

export interface IConfigurationData extends IConfigurationDataBase {
  SERVER_PORT: number;
  HOST: string;

  // Auth
  BASIC_AUTH_USERNAME: string;
  BASIC_AUTH_PASSWORD: string;

  // Auth Tokens
  AUTH_JWT_ISSUER: string;
  AUTH_WEB_KEY_HOST: string;

  // Infrastructure
  REDIS_PORT: number;

  MONGO_INITDB_ROOT_USERNAME: string;
  MONGO_INITDB_ROOT_PASSWORD: string;
  MONGO_HOST: string;
  MONGO_EXPOSE_PORT: number;
  MONGO_DB_NAME: string;
}

export class ConfigHandler extends ConfigurationBase<IConfigurationData> {
  constructor(options: IConfigurationOptions<IConfigurationData>) {
    super(options);
  }
}
