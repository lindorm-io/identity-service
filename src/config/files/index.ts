export * from "./development-config";
export * from "./environment-config";
export * from "./production-config";
export * from "./staging-config";
export * from "./test-config";
