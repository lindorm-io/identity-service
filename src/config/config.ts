import dotenv from "dotenv";
import { Audience } from "../enum";
import { ConfigHandler } from "./ConfigHandler";
import { MongoConnectionType } from "@lindorm-io/mongo";
import { NodeEnvironment } from "@lindorm-io/koa-config";
import { developmentConfig, environmentConfig, productionConfig, stagingConfig, testConfig } from "./files";

if (!process.env.NODE_ENV) dotenv.config();

const handler = new ConfigHandler({
  productionConfig,
  stagingConfig,
  developmentConfig,
  environmentConfig,
  testConfig,
});

export const { NODE_ENVIRONMENT } = environmentConfig;
const config = handler.get(process.env.NODE_ENV);

export const SERVER_PORT = config.SERVER_PORT;
export const HOST = config.HOST;

export const IS_TEST = NODE_ENVIRONMENT === NodeEnvironment.TEST;

export const BASIC_AUTH_MW_OPTIONS = {
  clients: [{ username: config.BASIC_AUTH_USERNAME, password: config.BASIC_AUTH_PASSWORD }],
};

export const BEARER_AUTH_MW_OPTIONS = {
  audience: Audience.ACCESS,
  issuer: config.JWT_ISSUER,
};

export const TOKEN_ISSUER_MW_OPTIONS = {
  issuer: config.JWT_ISSUER,
};

export const WEB_KEY_MW_OPTIONS = {
  host: config.WEB_KEY_HOST,
};

export const MONGO_CONNECTION_OPTIONS = {
  type: MongoConnectionType.STORAGE,
  auth: {
    user: config.MONGO_INITDB_ROOT_USERNAME,
    password: config.MONGO_INITDB_ROOT_PASSWORD,
  },
  url: {
    host: config.MONGO_HOST,
    port: config.MONGO_EXPOSE_PORT,
  },
  databaseName: config.MONGO_DB_NAME,
};
