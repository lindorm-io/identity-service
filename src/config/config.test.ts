import * as config from "./config";

describe("config", () => {
  test("should provide configuration", () => {
    expect(config).toMatchSnapshot();
  });
});
