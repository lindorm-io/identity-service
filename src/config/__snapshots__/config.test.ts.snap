// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config should provide configuration 1`] = `
Object {
  "BASIC_AUTH_MW_OPTIONS": Object {
    "password": "secret",
    "username": "secret",
  },
  "BEARER_AUTH_MW_OPTIONS": Object {
    "audience": "access",
    "issuer": "https://test.authentication.lindorm.io",
  },
  "HOST": "https://test.identity.lindorm.io",
  "IS_TEST": true,
  "MONGO_CONNECTION_OPTIONS": Object {
    "auth": Object {
      "password": "example",
      "user": "root",
    },
    "databaseName": "identity",
    "type": "STORAGE",
    "url": Object {
      "host": "localhost",
      "port": 27017,
    },
  },
  "NODE_ENVIRONMENT": "test",
  "SERVER_PORT": 3000,
  "TOKEN_ISSUER_MW_OPTIONS": Object {
    "issuer": "https://test.authentication.lindorm.io",
  },
  "WEB_KEY_MW_OPTIONS": Object {
    "host": "https://test.authentication.lindorm.io",
  },
}
`;
